lib/util/utils.go

```diff
@@ -178,22 +179,37 @@ func Address(network, host string) string {
 // AsService wraps the given function to implement suture.Service by calling
 // that function on serve and closing the passed channel when Stop is called.
 func AsService(fn func(stop chan struct{})) suture.Service {
-	return AsServiceWithError(func(stop chan struct{}) error {
+	return asServiceWithError(func(stop chan struct{}) error {
 		fn(stop)
 		return nil
 	})
 }
 
 type ServiceWithError interface {
 	suture.Service
+	fmt.Stringer
 	Error() error
 	SetError(error)
 }
 
 // AsServiceWithError does the same as AsService, except that it keeps track
 // of an error returned by the given function.
 func AsServiceWithError(fn func(stop chan struct{}) error) ServiceWithError {
+	return asServiceWithError(fn)
+}
+
+// caller retrieves information about the creator of the service, i.e. the stack
+// two levels up from itself.
+func caller() string {
+	pc := make([]uintptr, 1)
+	_ = runtime.Callers(4, pc)
+	f, _ := runtime.CallersFrames(pc).Next()
+	return f.Function
```

Comment:
This might need a few frames I think, as we have utilities that turn things into services, so the actual service creation is two frames up.

Reply:
I wrote a test to make sure it is actually 4 frames up, and that passes. If there's utility stuff on top of `lib/util`, we can't really know about that in lib/util, so we'd have to pass that information as an argument at which point we can just add `name string` to the arguments (but that would be tedious).

Reply:
Why would adding a name be tedious?

Reply:
I also think this is not tedious. :) `AsService(name string, fn func(stop chan struct{}))` seems perfectly cromulent

Reply:
Yeah, nothing tedious about the function itself, just tedious to change all invocations (though hopefully in most cases it will be simple as the caller hopefully implements `fmt.Stringer`). I'll maybe consider that along with a change to `context` instead of the stop channel.